File: src/main/java/j/flyweight/BigString.java
package j.flyweight;

import java.util.List;
import java.util.stream.Collectors;

public class BigString {
    private final List<BigChar> bigChars;

    public BigString(String string) {
        bigChars =
                string
                        .chars()
                        .mapToObj(c -> BigCharFactory.getInstance().getBigChar((char) c))
                        .collect(Collectors.toList());
    }

    public void print() {
        for (BigChar bc : bigChars) {
            bc.print();
        }
    }
}

File: src/main/kotlin/k/flyweight/BigString.kt
/* (C)2022 */
package k.flyweight

class BigString(string: String) {

  private val bigChars: List<BigChar> = string.map { BigCharFactory.getBigChar(it) }.toList()

  fun print() {
    for (bc in bigChars) bc.print()
  }
}
